zcash_client_backend/src/data_api.rs

```diff
@@ -236,12 +246,13 @@ pub trait WalletRead {
 /// decrypted and extracted from a [`CompactBlock`].
 ///
 /// [`CompactBlock`]: crate::proto::compact_formats::CompactBlock
-pub struct PrunedBlock<'a> {
+pub struct PrunedBlock<Nf> {
     pub block_height: BlockHeight,
     pub block_hash: BlockHash,
     pub block_time: u32,
-    pub commitment_tree: &'a sapling::CommitmentTree,
-    pub transactions: &'a Vec<WalletTx<sapling::Nullifier>>,
+    pub transactions: Vec<WalletTx<Nf>>,
+    pub sapling_commitment_tree_size: Option<u32>,
+    pub sapling_commitments: Vec<(sapling::Node, Retention<BlockHeight>)>,
```

Comment:
This will presumably need to add Orchard fields. Maybe add them now?

Reply:
This PR is already going to be a couple KLOC; Orchard should be a separate effort, not done piecemeal.